osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,395 @@
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, absolute_import, unicode_literals
+
+import json
+import logging
+import os
+import re
+from pkg_resources import parse_version
+
+from osbs.build.build_request import BuildRequest
+from osbs.build.spec import BuildUserParams
+from osbs.constants import (DEFAULT_OUTER_TEMPLATE, DEFAULT_INNER_TEMPLATE,
+                            DEFAULT_CUSTOMIZE_CONF, ISOLATED_RELEASE_FORMAT)
+from osbs.exceptions import OsbsException, OsbsValidationException
+from osbs.utils import (git_repo_humanish_part_from_uri,
+                        Labels)
+
+logger = logging.getLogger(__name__)
+
+_CM_SECRETS = ['required_secrets', 'worker_token_secrets']
+
+
+class BuildRequestV2(BuildRequest):
+    """
+    Wraps logic for creating build inputs
+    """
+
+    def __init__(self, build_json_store, inner_template=None,
+                 outer_template=None, customize_conf=None):
+        """
+        :param build_json_store: str, path to directory with JSON build files
+        :param inner_template: str, path to inner template JSON
+        :param outer_template: str, path to outer template JSON
+        :param customize_conf: str, path to customize configuration JSON
+        """
+        super(BuildRequestV2, self).__init__(build_json_store, inner_template,
+                                             outer_template, customize_conf)
+        self.spec = None
+        self.user_params = BuildUserParams()
+        self.build_json_store = build_json_store
+        self._inner_template_path = inner_template or DEFAULT_INNER_TEMPLATE
+        self._outer_template_path = outer_template or DEFAULT_OUTER_TEMPLATE
+        self._customize_conf_path = customize_conf or DEFAULT_CUSTOMIZE_CONF
+        self.build_json = None       # rendered template
+        self._template = None        # template loaded from filesystem
+        self._inner_template = None  # dock json
+        self._customize_conf = None  # site customize conf for _inner_template
+        self._dj = None
+        self._resource_limits = None
+        self._openshift_required_version = parse_version('1.0.6')
+        self._repo_info = None
+        # For the koji "scratch" build type
+        self.scratch = None
+        self.isolated = None
+        self.is_auto = None
+        self.base_image = None
+        self.scratch_build_node_selector = None
+        self.explicit_build_node_selector = None
+        self.auto_build_node_selector = None
+        self.isolated_build_node_selector = None
+        self.is_auto = None
+        # forward reference
+        self.platform_node_selector = None
+
+    # Override
+    def set_params(self, **kwargs):
+        """
+        set parameters according to specification
+
+        these parameters are accepted:
+
+        :param pulp_secret: str, resource name of pulp secret
```

Comment:
Let's update this comment with the user params/node_selectors we actually expect.